packages/davinci-client/src/lib/collector.utils.ts

```diff
@@ -20,11 +20,11 @@ import type { DaVinciField } from './davinci.types';
  * @param {ActionCollectorTypes} [collectorType] - Optional type of the ActionCollector.
  * @returns {ActionCollector} The constructed ActionCollector object.
  */
-export function returnActionCollector(
+export function returnActionCollector<CollectorType extends ActionCollectorTypes>(
   field: DaVinciField,
   idx: number,
-  collectorType?: ActionCollectorTypes,
-): ActionCollector {
+  collectorType: CollectorType,
```

Comment:
Can you elaborate on how this improves our typing?

Reply:
discussed on slack